modules/imap/output_modules.php

```diff
@@ -370,27 +370,31 @@ protected function output() {
             $txt .= ' | <a class="hlink" id="copy_message" href="#">'.$this->trans('Copy').'</a>';
             $txt .= ' | <a class="hlink" id="move_message" href="#">'.$this->trans('Move').'</a>';
             $txt .= ' | <a class="archive_link hlink" id="archive_message" href="#">'.$this->trans('Archive').'</a>';
-            $txt .= ' | ' . snooze_dropdown($this, isset($headers['X-Snoozed']));
+            if (!isset($headers['Flags']) || !stristr($headers['Flags'], 'draft')) {
+                $txt .= ' | ' . snooze_dropdown($this, isset($headers['X-Snoozed']));
+            }
             $txt .= ' | <a class="hlink" id="show_message_source" href="#">' . $this->trans('Show Source') . '</a>';
 
-            if ($this->get('sieve_filters_enabled')) {
-                $server_id = $this->get('msg_server_id');
-                $imap_server = $this->get('imap_accounts')[$server_id];
-                if ($this->get('sieve_filters_client')) {
-                    $user_config = $this->get('user_config');
-                    $contact_list = $user_config->get('contacts', []);
-                    $existing_emails = array_column($contact_list, 'email_address');
-                    $sender = addr_parse($headers['From'])['email'];
-                    $domain = '*@'.get_domain($sender);
-                    $blocked_senders = get_blocked_senders_array($imap_server, $this->get('site_config'), $this->get('user_config'));
-                    $sender_blocked = in_array($sender, $blocked_senders);
-                    $domain_blocked = in_array($domain, $blocked_senders);
-                    if(!in_array($sender, $existing_emails)){
-                        $txt .= ' | <div class="dropdown d-inline-block"><a class="block_sender_link hlink dropdown-toggle'.($domain_blocked || $sender_blocked ? '" id="unblock_sender" data-target="'.($domain_blocked? 'domain':'sender').'"' : '"').' href="#" aria-labelledby="dropdownMenuBlockSender" data-bs-toggle="dropdown"><i class="bi bi-lock-fill"></i> <span id="filter_block_txt">'.$this->trans($domain_blocked ? 'Unblock Domain' : ($sender_blocked ? 'Unblock Sender' : 'Block Sender')).'</span></a>';
-                        $txt .= block_filter_dropdown($this);
+            if (!isset($headers['Flags']) || !stristr($headers['Flags'], 'draft')) {
```

Comment:
this is uncessary you can add that condition to the if below ($this->get('sieve_filters_enabled'))
`if ($this->get('sieve_filters_enabled') && !$is_draft)`